tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,60 +159,104 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+        fluid.tests.verifyToolTips(pager, sequence, 2, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "cat"} ],
+            [ {nodeName: "b", nodeText: "bird"}, {nodeName: "b", nodeText: "fish"} ],
+            [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "dragon"} ],
+            [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+        fluid.tests.verifyToolTips(pager, sequence, 3, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "bird"} ],
+            [ {nodeName: "b", nodeText: "fish"}, {nodeName: "b", nodeText: "dragon"} ],
+            [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+        fluid.tests.verifyToolTips(pager, sequence, 4, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "fish"} ],
+            [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+        
+        return {
+            name: "Pager tooltip tests",
+            tests: [
+                {
+                    name: "Tooltip visibility and contents",
+                    sequence: sequence
                 }
-            });
+            ]
         };
+    };
 
-        var tooltipTest = function (location) {
-            return function (idx, linkEl) {
-                var linkId = linkEl.id;
-                var link = $(linkEl);
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "focus"
+    fluid.tests.assertVisibleTips = function (pager, sequence, message, targetIds, expectedTooltipContents, index) {
+        sequence.push({
+            event: "{trackTooltips}.events.notifyFocusChange",
+            listener: function () {
+                fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
+                    jqUnit.assertNode("When the page size is " + pager.model.pageSize + ", The contents of the tooltip should be set", expectedTooltipContents[index],
+                        $("b", tooltip));
                 });
+            }
+        });
+    };
 
-                if (link.hasClass(pager.pagerBar.options.styles.currentPage)) { // assumption that this will not change between now and then!
-                    assertVisibleTips("There shouldn't be any tooltips visible when the currentPage is focused", []);
-                } else {
-                    assertVisibleTips("Only the tooltip for page link " + (idx + 1) + ", in the " + location + " page bar is visible", [linkId]);
+    /*
+     * Change the page size if it has provided and
+     * verify whether the displaying links and their tooltips are synced are with page size
+     */
+    fluid.tests.verifyToolTips = function (pager, sequence, pageSize, expectedTooltipContent) {
+        if (pageSize) {
+            sequence.push({
```

Comment:
I think @cindyli mentioned this before, but I'm also having trouble following along all of the sequence generation parts. I wonder if you could make some of these more static and reuse through concatenation, grade merging, or the like. Also it isn't obvious from the function name and description that the function will actually modify the sequence you pass into it. This could cause some unintended side effects for someone using it.